modules/virt-using-wasp-agent-to-configure-higher-vm-workload-density.adoc

```diff
@@ -0,0 +1,298 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-using-wasp-agent-to-configure-higher-vm-workload-density_{context}"]
+= Using wasp-agent to configure higher VM workload density
+
+`wasp-agent` is a component that enables an OpenShift cluster to assign swap resources to VM workloads.
+Swap usage is supported on worker nodes only.
+
+[NOTE]
+====
+Swap resources can only be assigned to virtual machine workloads (VM pods) of the `Burstable` Quality of Service (QoS) class. VM pods of the `Guaranteed` QoS class and pods of any QoS class that do not belong to VMs cannot swap.
+
+For descriptions of QoS classes, see link:https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/[Configure Quality of Service for Pods]
+====
+
+.Prerequisites
+
+* The `oc` tool is available.
+* You are logged into the cluster with the cluster-admin role.
+* A memory over-commit ratio is defined. The default is 150%.
+* The node belongs to a worker pool.
+
+.Procedure
+
+[NOTE]
+====
+`wasp-agent` deploys an OCI hook to enable swap usage for containers on the node level. The low-level nature requires the `DaemonSet` object to be privileged.
+====
+
+. Create a privileged service account as follows:
++
+[source,terminal]
+----
+$ oc adm new-project wasp
+$ oc create sa -n wasp wasp
+$ oc adm policy add-cluster-role-to-user cluster-admin -n wasp -z wasp
+$ oc adm policy add-scc-to-user -n wasp privileged -z wasp
+----
+. Deploy `wasp-agent`:
+.. Create a `DaemonSet` object as follows:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: wasp-agent
+  namespace: wasp
+  labels:
+    app: wasp
+    tier: node
+spec:
+  selector:
+    matchLabels:
+      name: wasp
+  template:
+    metadata:
+      annotations:
+        description: >-
+          Configures swap for workloads
+      labels:
+          name: wasp
+    spec:
+      serviceAccountName: wasp
+      hostPID: true
+      hostUsers: true
+      terminationGracePeriodSeconds: 5
+      containers:
+        - name: wasp-agent
+          image: >-
+            registry.redhat.io/container-native-virtualization/wasp-agent-rhel9:latest
+          imagePullPolicy: Always
+          env:
+          - name: "FSROOT"
+            value: "/host"
+          resources:
+            requests:
+              cpu: 100m
+              memory: 50M
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - name: host
+            mountPath: "/host"
+      volumes:
+      - name: host
+        hostPath:
+          path: "/"
+      priorityClassName: system-node-critical
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+      maxSurge: 0
+status: {}
+----
+.. Get the latest URL from the link:https://registry.redhat.io/container-native-virtualization/wasp-agent-rhel9:latest[RH portal] and enter the URL in the created `DaemonSet` object.
```

Comment:
```suggestion
+
.. Get the latest URL from the link:https://registry.redhat.io/container-native-virtualization/wasp-agent-rhel9:latest[RH portal] and enter the URL in the created `DaemonSet` object.
```

Link does not work and renders to a 404.

Reply:
@dfitzmau It is supposed to work after the CNV 4.16 GA.